mozilla/subhub
# mypy pre-commit hook addition

This pull request (PR) provides support for the execution of mypy analysis as part of the pre-commit hook suite.

Steps to test:

1.  git fetch
2.  git checkout feature/my-precommit-hook
3.  pre-commit and observe the output

```
vagrant@ubuntu-xenial:/opt/subhub$ pre-commit
[WARNING] Unstaged files detected.
[INFO] Stashing unstaged files to /home/vagrant/.cache/pre-commit/patch1573142333.
Check for added large files..........................(no files to check)Skipped
Check JSON...........................................(no files to check)Skipped
Detect Private Key...................................(no files to check)Skipped
Fix End of Files.....................................(no files to check)Skipped
Pretty format JSON...................................(no files to check)Skipped
Trim Trailing Whitespace.............................(no files to check)Skipped
Check hooks apply to the repository..................(no files to check)Skipped
Check for useless excludes...........................(no files to check)Skipped
pylint...............................................(no files to check)Skipped
black................................................(no files to check)Skipped
mypy.................................................(no files to check)Skipped
[INFO] Restored changes from /home/vagrant/.cache/pre-commit/patch1573142333.
```

Notice the `mypy` line at the end of the output.


---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/mozilla/subhub/352)
